hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/LegacyReplicationManager.java

```diff
@@ -398,7 +400,21 @@ protected void processContainer(ContainerInfo container,
            */
           if (replicas.isEmpty() && (container.getNumberOfKeys() == 0)) {
             closeEmptyContainer(container);
+            return;
+          }
+
+          if (!foundHealthy) {
```

Comment:
I wonder if we should check the container is a RATIS container here too, or at least not EC? EC containers have no handling for Quasi-Closed, so letting an EC container go to Quasi_Closed may cause more problems.

Reply:
That does make sense, yes. I'll add a check.

Reply:
Updated in the latest commit.